sdk/servicebus/Azure.Messaging.ServiceBus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,309 @@
+
+## Logging and diagnostics
+
+The Service Bus client library is fully instrumented for logging information at various levels of detail using the .NET `EventSource` to emit information. Logging is performed for each operation and follows the pattern of marking the starting point of the operation, it's completion, and any exceptions encountered. Additional information that may offer insight is also logged in the context of the associated operation.
+
+### Enable logging
+
+The Service Bus client logs are available to any `EventListener` by opting into the sources starting with "Azure-Messaging-ServiceBus" or by opting into all sources that have the trait "AzureEventSource". To make capturing logs from the Azure client libraries easier, the `Azure.Core` library used by Service Bus offers an `AzureEventSourceListener`.
+
+For more information, see: [Logging with the Azure SDK for .NET][Logging].
+
+### Distributed tracing
+
+The Service Bus client library supports distributed tracing though integration with the Application Insights SDK. It also supports **experimental** support for the the OpenTelemetry specification via the .NET [ActivitySource][ActivitySource] type introduced in .NET 5. In order to enable `ActivitySource` support for use with OpenTelemetry, see [ActivitySource support][ActivitySourceSupport].
+
+In order to use the GA DiagnosticActivity support, you can integrate with the Application Insights SDK. More details can be found in [ApplicationInsights with Azure Monitor][AppInsights].
+
+The library creates the following spans:
+
+`Message`  
+`ServiceBusSender.Send`  
+`ServiceBusSender.Schedule`  
+`ServiceBusSender.Cancel`  
+`ServiceBusReceiver.Receive`  
+`ServiceBusReceiver.ReceiveDeferred`  
+`ServiceBusReceiver.Peek`  
+`ServiceBusReceiver.Abandon`  
+`ServiceBusReceiver.Complete`  
+`ServiceBusReceiver.DeadLetter`  
+`ServiceBusReceiver.Defer`  
+`ServiceBusReceiver.RenewMessageLock`  
+`ServiceBusSessionReceiver.RenewSessionLock`  
+`ServiceBusSessionReceiver.GetSessionState`  
+`ServiceBusSessionReceiver.SetSessionState`  
+`ServiceBusProcessor.ProcessMessage`  
+`ServiceBusSessionProcessor.ProcessSessionMessage`  
+`ServiceBusRuleManager.CreateRule`  
+`ServiceBusRuleManager.DeleteRule`  
+`ServiceBusRuleManager.GetRules`  
+
+Most of the spans are self-explanatory and are started and stopped during the operation that bears its name. The span that ties the others together is `Message`. The way that the message is traced is via the the `Diagnostic-Id` that is set in the [ServiceBusMessage.ApplicationProperties][ApplicationProperties] property by the library during send and schedule operations. In Application Insights, `Message` spans will be displayed as linking out to the various other spans that were used to interact with the message, e.g. the receive span, the send span, and the complete span would all be linked from the `Message` span. Here is an example of what this looks like in Application Insights:
```

Comment:
```suggestion
Most of the spans are self-explanatory and are started and stopped during the operation that bears its name. The span that ties the others together is `Message`. The way that the message is traced is via the the `Diagnostic-Id` that is set in the [ServiceBusMessage.ApplicationProperties][ApplicationProperties] property by the library during send and schedule operations. In Application Insights, `Message` spans will be displayed as linking out to the various other spans that were used to interact with the message, e.g. the `ServiceBusReceiver.Receive` span, the `ServiceBusSender.Send` span, and the `ServiceBusReceiver.Complete` span would all be linked from the `Message` span. Here is an example of what this looks like in Application Insights:
```

Just some formatting to make it more obvious these are terms, not generic words.

Reply:
Thank you!